linera-core/src/worker.rs

```diff
@@ -418,21 +417,23 @@ where
         &self,
         certificate: Certificate,
         blobs: Vec<Blob>,
-        mut notifications: Option<&mut impl Extend<Notification>>,
+        notifications: &impl NotificationSink<Notification>,
```

Comment:
I think the name `notifications` made more sense when this was known to be a collection of notifications. Now it's a generic `NotificationSink` maybe `notification_sink` makes more sense.

Reply:
Yes exactly. Or just `notifier`.

Reply:
I don't love `notifier` because we have a struct by that name right in this crate, which this may not be.